### ISSUE TITLE
Format exception when returning the result from RecognizeOnceAsync

### ISSUE THREAD

**Issue by ChrisProlls** [Azure-Samples/cognitive-services-speech-sdk#887]
**Describe the bug**
I followed this quickstart to enroll my voice and check my voice afterward in another audio stream : https://docs.microsoft.com/en-us/azure/cognitive-services/speech-service/get-started-speaker-recognition?tabs=script&pivots=programming-language-csharp.
I used the **text-independant** examples. I just changed the locale for the `CreateProfileAsync` method, I used fr-FR.
````
await client.CreateProfileAsync(VoiceProfileType.TextIndependentVerification, "fr-FR")
````
The enrollment seems to be ok, but when I try to recognize my voice with the `RecognizeOnceAsync` method, I get an `InvalidFormatException` when the result is fetched from the web service (see thr stack trace below for more info). The error occurs when the `Convert.ToDouble()` is called.

**To Reproduce**
Use the **text-independant** example from the link with fr-FR locale for the profil.

**Expected behavior**
No exception thrown

**Version of the Cognitive Services Speech SDK**
Microsoft.CognitiveServices.Speech: 1.14.0

**Platform, Operating System, and Programming Language**
 - OS: Windows
 - Hardware - x64
 - Programming language: C#
 - Azure : Azure Speech service in West US region

**Additional context**
Here is the stack trace
```
Input string was not in a correct format.
at System.Number.ThrowOverflowOrFormatException(ParsingStatus status, TypeCode type)
   at System.Convert.ToDouble(String value, IFormatProvider provider)
   at Microsoft.CognitiveServices.Speech.SpeakerRecognitionResult..ctor(IntPtr resultHandlePtr)
   at Microsoft.CognitiveServices.Speech.SpeakerRecognizer.<>c__DisplayClass5_0.<RecognizeOnceAsync>b__0(InteropSafeHandle h)
   at Microsoft.CognitiveServices.Speech.SpeakerRecognizer.<>c__DisplayClass8_0.<RunAsync>b__0()
   at System.Threading.Tasks.Task`1.InnerInvoke()
   at System.Threading.Tasks.Task.<>c.<.cctor>b__274_0(Object obj)
   at System.Threading.ExecutionContext.RunFromThreadPoolDispatchLoop(Thread threadPoolThread, ExecutionContext executionContext, ContextCallback callback, Object state)
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Threading.ExecutionContext.RunFromThreadPoolDispatchLoop(Thread threadPoolThread, ExecutionContext executionContext, ContextCallback callback, Object state)
   at System.Threading.Tasks.Task.ExecuteWithThreadLocal(Task& currentTaskSlot, Thread threadPoolThread)
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter`1.GetResult()
   at ConsoleApp1.Program.<SpeakerVerify>d__3.MoveNext() in D:\_Docs\Projets\ITCE\MultiLocuteur\ITCE.Multilocuteur.Console\ConsoleApp1\Program.cs:line 98
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.GetResult()
   at ConsoleApp1.Program.<VerificationEnroll>d__2.MoveNext() in D:\_Docs\Projets\ITCE\MultiLocuteur\ITCE.Multilocuteur.Console\ConsoleApp1\Program.cs:line 81
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.GetResult()
   at ConsoleApp1.Program.<Main>d__0.MoveNext() in D:\_Docs\Projets\ITCE\MultiLocuteur\ITCE.Multilocuteur.Console\ConsoleApp1\Program.cs:line 16
```

**Comment by BrianMouncer** [comment]
I see you are using french, but are you still using the WestUS data region.
The speaker recognition service is not available in all data centers.

Where you able to use the SDK sample as is, before switching to French?

https://github.com/Azure-Samples/cognitive-services-speech-sdk/tree/master/quickstart/csharp/dotnet/speaker-recognition

@trrwilson do you have have more to add, I know you've worked in this area before?

Thanks,

**Comment by ChrisProlls** [comment]
Hi @BrianMouncer, thanks for your reply.
I used the WestUS data region because it seems that this is the only avalaible region for Speaker recognition as the docs says here : https://docs.microsoft.com/en-us/azure/cognitive-services/speech-service/get-started-speaker-recognition#prerequisites

BUT, when I search for the documentation about Language support, I found that Speaker recognition support `fr-FR` : https://docs.microsoft.com/en-us/azure/cognitive-services/speech-service/language-support#speaker-recognition

**Comment by BrianMouncer** [comment]
Good afternoon @ChrisProlls

I am reviewing our open GitHub issues today, and was attempting to reproduce this locally, however I was unable to get this to happen using https://github.com/Azure-Samples/cognitive-services-speech-sdk/blob/master/quickstart/csharp/dotnet/speaker-recognition/README.md

I am using this sample project with a speech key from "westus" region.  First I verified the sample was working unchanged with en-us locale voice profile creation. Then I updated the CreateProfileAsync() calls to create a profile for "fr-fr" instead of "en-us".

The sample worked many many time, for both en-us and fr-fr locales.
Can you tell me know if you are still seeing this issue in the latest 1.15.0 release of the SDK?
If so, do you know a better way to reliably reproduce the problem?

Thanks.

**Comment by lnstreit** [comment]
Hello,

As @ChrisProlls, I encounter the same issue on my side. Did you manage to resolve the issue now (if yes how) ?

**Bug:** 
I downloaded the solution files found here: https://github.com/Azure-Samples/cognitive-services-speech-sdk/tree/master/quickstart/csharp/dotnet/speaker-recognition
The only changes I made to the code was to change "YourSubscriptionKey" and "YourServiceRegion" with my cognitive services speech key and "westus" as it is the only region available for speaker recognition at the time being.
As @ChrisProlls, the enrolment of the profiles seems okay, but the bug occurs when the script tries to verify or identify the input voice when the `System.Number.ParseDouble()` function is called (I assume the different function is due to the fact that I use SDK version 1.15.0 instead of 1.14.0).

**SDK version:**
1.15.0

**Platform, Operating System, and Programming Language:**
OS: Windows 10
Hardware - x64
Programming language: C# .NET
Azure : Azure Speech service in West US region

**Stack trace:**
```
System.FormatException
  HResult=0x80131537
  Message=Input string was not in a correct format.
  Source=mscorlib
  StackTrace:
   at System.Number.ParseDouble(String value, NumberStyles options, NumberFormatInfo numfmt)
   at Microsoft.CognitiveServices.Speech.SpeakerRecognitionResult..ctor(IntPtr resultHandlePtr)
   at Microsoft.CognitiveServices.Speech.SpeakerRecognizer.<>c__DisplayClass6_0.<RecognizeOnceAsync>b__0(InteropSafeHandle h)
   at Microsoft.CognitiveServices.Speech.SpeakerRecognizer.<>c__DisplayClass8_0.<RunAsync>b__0()
   at System.Threading.Tasks.Task`1.InnerInvoke()
   at System.Threading.Tasks.Task.Execute()
--- End of stack trace from previous location ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter`1.GetResult()
   at helloworld.Program.<IdentifySpeakersAsync>d__1.MoveNext() in C:\Users\ltr\Documents\AI ML Tests\Azure\cognitive-services-speech-sdk\quickstart\csharp\dotnet\speaker-recognition\helloworld\Program.cs:line 40
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.GetResult()
   at helloworld.Program.<SpeakerIdentificationAsync>d__3.MoveNext() in C:\Users\ltr\Documents\AI ML Tests\Azure\cognitive-services-speech-sdk\quickstart\csharp\dotnet\speaker-recognition\helloworld\Program.cs:line 98
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at helloworld.Program.<SpeakerIdentificationAsync>d__3.MoveNext() in C:\Users\ltr\Documents\AI ML Tests\Azure\cognitive-services-speech-sdk\quickstart\csharp\dotnet\speaker-recognition\helloworld\Program.cs:line 104
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.GetResult()
   at helloworld.Program.<Main>d__5.MoveNext() in C:\Users\ltr\Documents\AI ML Tests\Azure\cognitive-services-speech-sdk\quickstart\csharp\dotnet\speaker-recognition\helloworld\Program.cs:line 174
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.GetResult()
   at helloworld.Program.<Main>(String[] args)

  This exception was originally thrown at this call stack:
    System.Number.ParseDouble(string, System.Globalization.NumberStyles, System.Globalization.NumberFormatInfo)
    Microsoft.CognitiveServices.Speech.SpeakerRecognitionResult.SpeakerRecognitionResult(System.IntPtr)
```

Thanks in advance for any help!

**Comment by glecaros** [comment]
Hi @lnstreit,

Can you please capture some logs ([instructions](https://docs.microsoft.com/en-us/azure/cognitive-services/speech-service/how-to-use-logging#sample)) and share them?

**Comment by lnstreit** [comment]
Hi all,

I resolved the issue in the end. The problem was due to the difference in culture formats between the speech configuration (or cognitive services) and the local machine (mine). As I use a machine which formats strings in the "fr-fr" format, doubles in strings are probably used with "," and not "." as in "en-us". The below line of code resolved the issue (changing the system format to the right one):

`System.Threading.Thread.CurrentThread.CurrentCulture = System.Globalization.CultureInfo.GetCultureInfo("en-us");`

Thanks for your will to help @glecaros!
